src/docs/self-hosted/productionalizing.mdx

```diff
@@ -0,0 +1,219 @@
+---
+title: 'Self-Hosted Productionalizing'
+---
+
+This page provides a guide to help users secure their Sentry deployments and make the most out of Sentry. For further guide relating system security and performance, you might want to 
+
+## Recommended system resource
+
+Depending on your usage, required system resource to run Sentry may varies. Minimum system specification for running Sentry looks like this:
+
+- 2 CPU cores
+- 4 GB RAM
+
+Having a machine that have lower system specification than that will fail the installation script.
+
+Current recommended system specification for running Sentry is:
+
+- 4 CPU cores
+- 16 GB RAM
+- 20 GB free disk storage space
+
+Depending on your load, you might want to increase your system specification to handle higher traffic load better. If increasing the disk storage space isn't possible, you can migrate your storage to use external storage such as AWS S3 or Google Cloud Storage (GCS). Decreasing your `SENTRY_RETENTION_DAYS` environment variable to lower numbers will save some storage space from being full, at the cost of having shorter data retention period.
```

Comment:
I really wanted to add how they can actually use Kubernetes for scaling out, but that's not officially supported. Is it okay to put it here as "community stuff"? 

A few stuff that I want to put here:
- https://github.com/sentry-kubernetes/charts
- https://github.com/kanadaj/sentry-operator

Reply:
@chadwhitacre what is the company policy on doing this? I would refrain from putting specific links to repos that aren't `getsentry` repos if we can help it.

Reply:
Yeah we need to draw the line somewhere and this seems like a good place to draw it @hubertdeng123.